### PR TITLE
Fixes to git and build environment

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
add .gitignore file for bootstrapping and built files
update csproj to not use deprecated nupkg
fix build.ps1 to correctly copy psd1 and psm1 to output folder

### CODE REVIEW

**Review by alerickson — APPROVED** [review]

File: src/build.ps1 [hunk 1]
```diff
@@ -74,38 +71,44 @@ write-host ("solutionDir: $($solutionDir)")
...
 )
 
-
-
-$itemsToCopyBinaries = $assemblyNames | % { "$solutionDir\$_\bin\$Configuration\$currentFramework\$_.dll" }
-$itemsToCopyPdbs = $assemblyNames | % { "$solutionDir\$_\bin\$Configuration\$currentFramework\$_.pdb" }
-
-$itemsToCopyCommon = @("$solutionDir\PowerShellGet.psd1")
+$projectPath = Split-Path $solutionDir -Parent
+$itemsToCopyCommon = @("$projectPath/PowerShellGet.psd1","$projectPath/PSModule.psm1")
```
Comment (TylerLeonhardt) [inline src/build.ps1]:
```suggestion
$itemsToCopyCommon = @(
    (Join-Path $projectPath "PowerShellGet.psd1")
    (Join-Path $projectPath "PSModule.psm1")
)
```

I know yours works, but [a wise man once said](https://twitter.com/Steve_MSFT/status/1302978236226232321) this is best practice :)

**Review by TylerLeonhardt — APPROVED** [review]

File: src/build.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply src/build.ps1]: lol, fair enough

File: src/PowerShellGet.csproj [hunk 2]
```diff
@@ -18,7 +18,6 @@
     <PackageReference Include="NuGet.Packaging" Version="5.7.0-rtm.6702" />
     <PackageReference Include="NuGet.ProjectModel" Version="5.7.0-rtm.6702" />
     <PackageReference Include="NuGet.Protocol" Version="5.7.0-rtm.6702" />
-    <PackageReference Include="NuGet.Protocol.Core.Types" Version="4.3.0-beta2-2463" />
```
Comment (anamnavi) [inline src/PowerShellGet.csproj:21]: why is this package reference being removed from the .csproj file? if it's deprecated would we need to use a later version (i.e Version="4.3.0-beta1-2418") (don't have an issue with it per se, just asking for my understanding!)

**Review by anamnavi — APPROVED** [review]
Looks good to me, just commented a question for my understanding. Thanks!

File: src/PowerShellGet.csproj (hunk 2 above)
Comment (alerickson) [reply src/PowerShellGet.csproj:21]: I had the same question, but when I pulled these changes and tried building it built just fine without that project reference.  Now that I think of it though, could it be that that the package was already on my machine so it was able to still use it?

Comment (SteveL-MSFT) [reply src/PowerShellGet.csproj:21]: I believe NuGet.Protocol is sufficient as manual testing using publish-psresource still works w/o that pkg

Comment (alerickson) [reply src/PowerShellGet.csproj:21]: It's referenced in Find and Install as well

Comment (SteveL-MSFT) [reply src/PowerShellGet.csproj:21]: Validated that Find-PSResource also works.  Having some other issue with Install-PSResource unrelated to this change (fails with beta9).

**Pull request merged**